src/session/session_compact.c

```diff
@@ -334,6 +334,17 @@ __wt_session_compact(WT_SESSION *wt_session, const char *uri, const char *config
     session = (WT_SESSION_IMPL *)wt_session;
     SESSION_API_CALL(session, compact, config, cfg);
 
+    /* Trigger the background server thread and toggle connection statistic. */
+    if ((ret = __wt_config_gets(session, &cfg[1], "background", &cval) == 0)) {
+        /* TODO: Check background compaction server running status and trigger the mutex. */
+        if (cval.val) {
+            WT_STAT_CONN_SET(session, session_background_compact_running, 1);
```

Comment:
I think this stat should be done as part of #9456. In my opinion, the thread that manage background compaction should update that stat as soon as its running.

Reply:
Good point

Reply:
I've removed the stat entirely as s_all won't pass unless it's used.